superset/db_engine_specs.py

```diff
@@ -74,6 +84,82 @@ def extra_table_metadata(cls, database, table_name, schema_name):
         """Returns engine-specific table metadata"""
         return {}
 
+    @staticmethod
+    def csv_to_df(**kwargs):
+        kwargs['filepath_or_buffer'] = \
+            config['UPLOAD_FOLDER'] + kwargs['filepath_or_buffer']
+        kwargs['encoding'] = 'utf-8'
+        kwargs['iterator'] = True
+        chunks = pandas.read_csv(**kwargs)
+        df = pandas.DataFrame()
+        df = pandas.concat(chunk for chunk in chunks)
+        return df
+
+    @staticmethod
+    def df_to_db(**kwargs):
+        df = kwargs['df']
+        table = kwargs['table']
+        del kwargs['df']
+        del kwargs['table']
+        kwargs['con'] = create_engine(kwargs['con'], echo=False)
+        df.to_sql(**kwargs)
+
+        table.user_id = g.user.id
+        table.schema = kwargs['schema']
+        db.session.add(table)
+        db.session.commit()
+
+    @staticmethod
+    def upload_csv(form, table):
+        def allowed_file(filename):
+            # Only allow specific file extensions as specified in the config
+            extension = os.path.splitext(filename)[1]
+            return extension and extension[1:] in config['ALLOWED_EXTENSIONS']
+
+        filename = secure_filename(form.csv_file.data.filename)
+        if not allowed_file(filename):
+            flash("Invalid file type selected.", 'alert')
+            return False
+        kwargs = {
```

Comment:
I don't think I was super clear about the kwargs comment earlier, and the verbose boilerplate isn't that bad either (others may consider the verbose, metaprogramming-free option better). 

Knowing the form object has a data dict:
http://wtforms.simplecodes.com/docs/0.6.1/forms.html#wtforms.form.Form.data

You can do something like:
```
form_data = form.data
csv_to_df_fields = ['names', 'filepath_or_buffer', ...]
kwargs = {s: form_data.get(s) for s in csv_to_df_fields}
kwargs['chunksize'] = 10000
# other alterations as needed
df = BaseEngineSpec.csv_to_df(**kwargs)
```